### PR TITLE
fix: do not use `unload` events in Chromium from App Runner

### LINKED ISSUE #29880: [Cypress 13.13.1, Chrome] - unload is not allowed
### Current behavior

image

image

### Desired behavior

No extra error in Chrome DevTools Console

### Test code to reproduce

I have this in my cypress.config.js file

```js
on('before:browser:launch', (_, launchOptions) => {
    launchOptions.args.push('--enable-features=DeprecateUnload')
    return launchOptions
});
```

### Cypress Version

13.13.1

### Node version

v20.15.1

### Operating System

Windows 10.0.19045

### Debug Logs

_No response_

### Other

_No response_

### PR DESCRIPTION
**Pull request by cacieprins**
- Closes https://github.com/cypress-io/cypress/issues/29880

### Additional details
At the point when the Runner's `EventManager` adds global event listeners, `setup()` has yet to be called, so there is not an instance of Cypress to use for the browser family check. To fix, `getRunnerConfigFromWindow()` is extracted into its own file so it can be imported by both `src/runner/index.ts` and `src/runner/event-manager.ts` without causing a circular dependency.

### Steps to test

Run a simple test in open mode with the following Cypress config:

```js
module.exports = {
  e2e: {
    setupNodeEvents(on, config) {
      on('before:browser:launch', (_, launchOptions) => {
        launchOptions.args.push('--enable-features=DeprecateUnload')

        return launchOptions
      })
      return config
    },
  },
}
```

Observe that there are no warnings in the browser console about a disallowed `unload` event.

This behavior cannot be tested with Cy in Cy tests, as far as I can tell, and there are no unit tests for `app`, so this fix is currently only manually tested.

### PR Tasks

- [NA] Have tests been added/updated?
- [NA] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [NA] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

File: packages/app/src/runner/event-manager.ts [hunk 1]
```diff
@@ -400,10 +401,8 @@ export class EventManager {
     this._addListeners()
   }
 
-  isBrowser (browserName) {
-    if (!this.Cypress) return false
-
-    return this.Cypress.isBrowser(browserName)
+  isBrowserFamily (family) {
+    return getRunnerConfigFromWindow()?.browser?.family === family
```
Comment (jennifer-shehane) [inline packages/app/src/runner/event-manager.ts]: This function is expecting a string, but it's being passed an object so this won't ever compare correctly.

**Review by jennifer-shehane — CHANGES_REQUESTED** [review]

File: packages/app/src/runner/event-manager.ts (hunk 1 above)
Comment (cacieprins) [reply packages/app/src/runner/event-manager.ts]: I'd waffled too many times between an obj and a string parameter - adding type declaration to parameter will help catch this in the future. addressed in https://github.com/cypress-io/cypress/pull/30061/commits/4af0b991cebd18abba4111220e1e6f0c3b3836f5

File: packages/driver/patches/jquery+3.4.1.dev.patch [hunk 2]
```diff
@@ -89,7 +89,7 @@ index 773ad95..84ca2f6 100644
 +	// IE/Edge sometimes throw a "Permission denied" error when strict-comparing
 +	// two documents; shallow comparisons work.
 +	// eslint-disable-next-line eqeqeq
-+	if ( preferredDoc != document &&
++	if ( docElem.msMatchesSelector && preferredDoc != document &&
```
Comment (AtofStryker) [inline packages/driver/patches/jquery+3.4.1.dev.patch:92]: is this still relevant? Mostly curious how you tested this.

**Review by AtofStryker — APPROVED** [review]

**Review by jennifer-shehane — APPROVED** [review]
I'm not seeing the unload warning displaying from our own code anymore. I still see one coming from jQuery but will have to test with the built binary I guess since that's patched.

File: packages/driver/patches/jquery+3.4.1.dev.patch (hunk 2 above)
Comment (cacieprins) [reply packages/driver/patches/jquery+3.4.1.dev.patch:92]: This is basically a cherry-picked patch from jQuery 3.7.1. when manually testing this in chrome, I was still receiving unload errors coming from jquery without this change

**Pull request merged**